src/main/java/com/fasterxml/jackson/databind/util/JSONPObject.java

```diff
@@ -66,6 +66,8 @@ public void serializeWithType(JsonGenerator jgen, SerializerProvider provider, T
     public void serialize(JsonGenerator jgen, SerializerProvider provider)
             throws IOException, JsonProcessingException
     {
+        // Escape line-separator characters that break JSONP
+        jgen.setCharacterEscapes(JsonpCharacterEscapes.instance());
```

Comment:
This is not safe: generator may have existing escapes user has specified, and this would just replace them.